frontend-web/webclient/app/UserSettings/ChangeEmailSettings.tsx

```diff
@@ -0,0 +1,336 @@
+import {useCloudCommand} from "Authentication/DataHook";
+import {Client} from "Authentication/HttpClientInstance";
+import * as React from "react";
+import {useCallback, useEffect, useRef} from "react";
+import {Box, Button, Checkbox, Input, Label} from "ui-components";
+import * as Heading from "ui-components/Heading";
+import {snackbarStore} from "Snackbar/SnackbarStore";
+import {EmailSettings, retrieveEmailSettings, toggleEmailSettings} from "UserSettings/api";
+import {bulkRequestOf} from "DefaultObjects";
+import {file} from "UCloud";
+import stat = file.stat;
+
+interface UserDetailsState {
+    settings: EmailSettings
+}
+
+export const defaultEmailSettings: EmailSettings = {
+    newGrantApplication: true,
+    grantAutoApprove: true,
+    grantApplicationUpdated: true,
+    grantApplicationApproved: true,
+    grantApplicationRejected: true,
+    grantApplicationWithdrawn: true,
+    newCommentOnApplication: true,
+    applicationTransfer: true,
+    applicationStatusChange: true,
+    //Project
+    projectUserInvite: true,
+    projectUserRemoved: true,
+    verificationReminder: true,
+    userRoleChange: true,
+    userLeft: true,
+    lowFunds: true
+}
+
+export enum MailType {
+    NEW_GRANT_APPLICATION,
+    GRANT_AUTO_APPROVE,
+    GRANT_APPLICATION_UPDATED,
+    GRANT_APPLICATION_APPROVED,
+    GRANT_APPLICATION_REJECTED,
+    GRANT_APPLICATION_WITHDRAWN,
+    NEW_COMMENT_ON_APPLICATION,
+    APPLICATION_TRANSFER,
+    APPLICATION_STATUS_CHANGE,
+    //Project
+    PROJECT_USER_INVITE,
+    PROJECT_USER_REMOVED,
+    VERIFICATION_REMINDER,
+    USER_ROLE_CHANGE,
+    USER_LEFT,
+    LOW_FUNDS
+}
+
+const initialState: UserDetailsState = {
+    settings: defaultEmailSettings
+};
+
+type Action<T, B> = {type: T; payload: B};
+type UpdatePlaceholdersEmailSettings = Action<"UpdatePlaceholdersEmailSettings", UserDetailsState>;
+
+const reducer = (state: UserDetailsState, action: UpdatePlaceholdersEmailSettings): UserDetailsState => {
+    switch (action.type) {
+        case "UpdatePlaceholdersEmailSettings":
+            return {...state, ...action.payload};
+    }
+};
+
+export const ChangeEmailSettings: React.FunctionComponent<{setLoading: (loading: boolean) => void}> = () => {
+    const [commandLoading, invokeCommand] = useCloudCommand();
+    const [state, dispatch] = React.useReducer(reducer, initialState, () => initialState);
+
+    var currentEmailSettings: EmailSettings = defaultEmailSettings
```

Comment:
You are assigning `currentEmailSettings` to equal `defaultEmailSettings` at every component update. Is this intended?